mkt/websites/management/commands/import_games_from_csv.py

```diff
@@ -0,0 +1,138 @@
+# docs.google.com/spreadsheets/d/
+# 1tShUCIR3tilh2WGeoXQMjy4ExGf94_7MilZqorjPfrM/edit#gid=0
+import csv
+from os.path import basename, splitext
+
+from django.core.management.base import BaseCommand, CommandError
+from django.core.validators import URLValidator, ValidationError
+from django.db.transaction import atomic
+
+from mkt.constants.applications import DEVICE_DESKTOP
+from mkt.constants.base import STATUS_PUBLIC
+from mkt.tags.models import Tag
+from mkt.websites.models import Website
+from mkt.websites.tasks import fetch_icon, fetch_promo_imgs
+
+
+class ParsingError(Exception):
+    pass
+
+
+class Command(BaseCommand):
+    """
+    Usage:
+
+        python manage.py import_games_from_csv <file>
+
+    Batch import games from a CSV. If there is a validation error on anything,
+    nothing will be created. All-or-nothing.
+
+    Columns are:
+        - Name
+        - URL
+        - Description
+        - Icon URL: direct link to image.
+        - Promo Image URL: direct link to image, 1050x300px minimum.
+    """
+    help = u'Import Desktop Games from a CSV file'
+    args = u'<file>'
+    subcommand = splitext(basename(__file__))[0]
+
+    def clean_string(self, s):
+        return s.strip().decode('utf-8')
+
+    def validate_url(self, url):
+        if url:
+            URLValidator()(url)
+
+    def set_tags(self, instance, row):
+        keywords = [
+            'featured-game',
+            'featured-game-' + self.clean_string(row['Category']).lower()]
+        for keyword in keywords:
+            tag, _ = Tag.objects.get_or_create(tag_text=keyword)
+            instance.keywords.add(tag)
+
+    def set_url(self, instance, row):
+        url = self.clean_string(row['URL']).lower()
+        try:
+            self.validate_url(url)
+        except ValidationError:
+            raise ParsingError(u'%s has invalid URL %s' % (row['Name'], url))
+        instance.url = url
+
+    def set_icon(self, instance, row):
+        icon_url = self.clean_string(row['Icon URL'])
+        if icon_url:
+            self.validate_url(icon_url)
+            icon = fetch_icon(instance.pk, icon_url)
+            if not icon:
+                raise ValidationError('Icon for %s could not be fetched.' %
+                                      instance.name)
+        else:
+            raise ParsingError(u'%s has no icon' % (row['Name'],))
+
+    def set_promo_imgs(self, instance, row):
+        promo_img_url = self.clean_string(row['Promo Img URL'])
+        if promo_img_url:
+            self.validate_url(promo_img_url)
+            promo_img = fetch_promo_imgs(instance.pk, promo_img_url)
+            if not promo_img:
+                raise ValidationError('Promo img for %s failed fetching.' %
```

Comment:
Actually, I don't want any catch anything. If anything fails, I want to know so that I can go clean up the data.